Makefile

```diff
@@ -220,6 +220,12 @@ build-operator-image: ## Build the operator image.
 	--build-arg VCS_REF=$(VCS_REF) --build-arg VCS_URL=$(VCS_URL) \
 	--build-arg GOARCH=$(LOCAL_ARCH) -f Dockerfile .
 
+build-operator-dev-image: ## Build the operator dev image.
```

Comment:
I think it would be more flexible to have `build-operator-dev-image`, when users/developers what to retag the image after building images.

Reply:
they would still be able to if combined. Just need to overwrite the variables, e.g.
```
make build-operator-dev-image DEV_REGISTRY=example-registry OPERATOR_IMAGE_NAME=example-name VERSION=dev
```
Would be even easier to retag this way since it's just built with specified tag.

Reply:
I don't see the reason why we should combine the command. There is no difference for the end-users and separating can keep the flexiblity.

Reply:
It's a small quality of life thing. Most development workflow would look like:
```
make build-operator-dev-image
make build-push-dev-image
```
There would not be many scenarios where re-tagging the image is necessary since it's just the developer testing one thing. So if we combine the commands, developer would only need to run one command (for most scenarios):
```
make build-operator-dev-image
```
And the command would just be
```
	@docker build -t $(REGISTRY)/$(OPERATOR_IMAGE_NAME):$(VERSION)\
	--build-arg VCS_REF=$(VCS_REF) --build-arg VCS_URL=$(VCS_URL) \
	--build-arg GOARCH=$(LOCAL_ARCH) -f Dockerfile .
	@docker push $(REGISTRY)/$(OPERATOR_IMAGE_NAME):$(VERSION)
```

If developer needs to re-tag the image to something else, then they could still run the combined command:
```
make build-operator-dev-image REGISTRY=my-registry OPERATOR_IMAGE_NAME=new-name VERSION=new-version
```

If the command was not combined, they would have to run:
```
make build-operator-dev-image
docker tag operator:tag registry/operator:tag
docker push registry/operator:tag
```

Reply:
Currently, the developer just need to run one command
```
make build-push-dev-image
```

Please check line 237
```
build-push-dev-image: build-operator-dev-image  ## Build and push the operator dev images.
```

Reply:
my bad, I missed that